### ISSUE TITLE
Issue: user_interaction in jamfpro_policy is broken in 0.6.0

### ISSUE THREAD

**Issue by ecanault** [deploymenttheory/terraform-provider-jamfpro#499]
Hi,

Something seems to have changed between 0.5.1 and 0.6.0 in the `payloads.user_interaction` schema of `jamfpro_policy` resources. 

The problem is it seems to break the provider:

```
module.policies["771149E1-AEB0-4F35-BBE1-97CE52ED2212"].jamfpro_policy.object: Modifying... [id=262]
╷
│ Error: Plugin did not respond
│ 
│   with module.policies["771149E1-AEB0-4F35-BBE1-97CE52ED2212"].jamfpro_policy.object,
│   on terraform_modules/policies/resource_jamfpro_policies.tf line 1, in resource "jamfpro_policy" "object":
│    1: resource "jamfpro_policy" "object" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain
│ more details.
╵

Stack trace from the terraform-provider-jamfpro_v0.6.0 plugin:

panic: interface conversion: interface {} is nil, not bool

goroutine 97 [running]:
github.com/deploymenttheory/terraform-provider-jamfpro/internal/resources/policies.constructPayloadUserInteraction(0x1400059c4d0?, 0x140004e4008)
	github.com/deploymenttheory/terraform-provider-jamfpro/internal/resources/policies/constructor.go:542 +0x3a8
github.com/deploymenttheory/terraform-provider-jamfpro/internal/resources/policies.constructPayloads(0x14000925000, 0x140004e4008)
	github.com/deploymenttheory/terraform-provider-jamfpro/internal/resources/policies/constructor.go:294 +0x78
github.com/deploymenttheory/terraform-provider-jamfpro/internal/resources/policies.construct(0x14000925000)
	github.com/deploymenttheory/terraform-provider-jamfpro/internal/resources/policies/constructor.go:27 +0x64
github.com/deploymenttheory/terraform-provider-jamfpro/internal/resources/common.Update[...]({0x103c3a618, 0x1400058d340}, 0x14000925000, {0x103c1c000, 0x1400065e170}, 0x103c1f770, 0x140004f0190, 0x103c1f7c0)
	github.com/deploymenttheory/terraform-provider-jamfpro/internal/resources/common/crud.go:117 +0x120
github.com/deploymenttheory/terraform-provider-jamfpro/internal/resources/policies.update({0x103c3a618, 0x1400058d340}, 0x14000925000, {0x103c1c000, 0x1400065e170})
	github.com/deploymenttheory/terraform-provider-jamfpro/internal/resources/policies/crud.go:53 +0xa8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0x14000647420, {0x103c3a570, 0x14000a11530}, 0x14000925000, {0x103c1c000, 0x1400065e170})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.35.0/helper/schema/resource.go:840 +0xe4
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x14000647420, {0x103c3a570, 0x14000a11530}, 0x1400076fe10, 0x14000924e80, {0x103c1c000, 0x1400065e170})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.35.0/helper/schema/resource.go:947 +0x670
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x14000440000, {0x103c3a570?, 0x14000a11470?}, 0x1400025e870)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.35.0/helper/schema/grpc_provider.go:1155 +0xaa4
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x140005aa280, {0x103c3a570?, 0x14000a10a80?}, 0x1400058c150)
	github.com/hashicorp/terraform-plugin-go@v0.25.0/tfprotov5/tf5server/server.go:865 +0x2b4
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x103be6f20, 0x140005aa280}, {0x103c3a570, 0x14000a10a80}, 0x14000924180, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.25.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:611 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x140001c7400, {0x103c3a570, 0x14000a109f0}, {0x103c40c00, 0x140005be340}, 0x1400020c000, 0x140005983c0, 0x1045340b8, 0x0)
	google.golang.org/grpc@v1.67.1/server.go:1394 +0xb78
google.golang.org/grpc.(*Server).handleStream(0x140001c7400, {0x103c40c00, 0x140005be340}, 0x1400020c000)
	google.golang.org/grpc@v1.67.1/server.go:1805 +0xb20
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.67.1/server.go:1029 +0x8c
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 49
	google.golang.org/grpc@v1.67.1/server.go:1040 +0x13c

Error: The terraform-provider-jamfpro_v0.6.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

I noticed that there was a small syntax change but I don't know if it's related to this issue: the `allow_user_to_defer` changed to `allow_users_to_defer` with an 's' at 'users'.

If I remove the `user_interaction` block from my code, it works as expected.

Regards, 
Emmanuel

**Comment by ShocOne** [comment]
hi @ecanault 

fix has been added to v0.6.1

**Comment by ecanault** [comment]
Hmmm...

It appears that something was fixed, but another part has been broken.

```
Stack trace from the terraform-provider-jamfpro_v0.6.1 plugin:

panic: interface conversion: interface {} is bool, not string

goroutine 138 [running]:
github.com/deploymenttheory/terraform-provider-jamfpro/internal/resources/macosconfigurationprofilesplist.constructMacOSConfigurationProfileSubsetSelfService(0x14000bb2f90)
	github.com/deploymenttheory/terraform-provider-jamfpro/internal/resources/macosconfigurationprofilesplist/constructor.go:180 +0x3d0
github.com/deploymenttheory/terraform-provider-jamfpro/internal/resources/macosconfigurationprofilesplist.constructJamfProMacOSConfigurationProfilePlist(0x14000acc980)
	github.com/deploymenttheory/terraform-provider-jamfpro/internal/resources/macosconfigurationprofilesplist/constructor.go:40 +0x4f0
github.com/deploymenttheory/terraform-provider-jamfpro/internal/resources/macosconfigurationprofilesplist.resourceJamfProMacOSConfigurationProfilesPlistUpdate({0x106078bb8, 0x14000a77180}, 0x14000acc980, {0x10605a340, 0x140006c2540})
	github.com/deploymenttheory/terraform-provider-jamfpro/internal/resources/macosconfigurationprofilesplist/crud.go:98 +0x104
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0x14000694620, {0x106078b10, 0x140008fe2d0}, 0x14000acc980, {0x10605a340, 0x140006c2540})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.35.0/helper/schema/resource.go:840 +0xe4
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x14000694620, {0x106078b10, 0x140008fe2d0}, 0x1400108b450, 0x14000acc800, {0x10605a340, 0x140006c2540})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.35.0/helper/schema/resource.go:947 +0x670
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x1400070c000, {0x106078b10?, 0x140008fe210?}, 0x14000a08140)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.35.0/helper/schema/grpc_provider.go:1155 +0xaa4
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x140005a21e0, {0x106078b10?, 0x14000e72e10?}, 0x14000a76150)
	github.com/hashicorp/terraform-plugin-go@v0.25.0/tfprotov5/tf5server/server.go:865 +0x2b4
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x106025200, 0x140005a21e0}, {0x106078b10, 0x14000e72e10}, 0x14000acc080, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.25.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:611 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x140001c7400, {0x106078b10, 0x14000e72a20}, {0x10607f1a0, 0x140006d8820}, 0x14000bbca20, 0x14000590240, 0x106978c58, 0x0)
	google.golang.org/grpc@v1.67.1/server.go:1394 +0xb78
google.golang.org/grpc.(*Server).handleStream(0x140001c7400, {0x10607f1a0, 0x140006d8820}, 0x14000bbca20)
	google.golang.org/grpc@v1.67.1/server.go:1805 +0xb20
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.67.1/server.go:1029 +0x8c
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 52
	google.golang.org/grpc@v1.67.1/server.go:1040 +0x13c

Error: The terraform-provider-jamfpro_v0.6.1 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```
